packages/collector/test/tracing/protocols/graphql/mockVersion.js

```diff
@@ -0,0 +1,45 @@
+/*
+ * (c) Copyright IBM Corp. 2022
+ * (c) Copyright Instana Inc. and contributors 2022
+ */
+
+'use strict';
+
+const semver = require('semver');
+const path = require('path');
+const mock = require('mock-require');
+const requireHook = require('../../../../../core/src/util/requireHook');
+
+const graphqlMajorDefault = semver.major(require(`${path.dirname(require.resolve('graphql'))}/package.json`).version);
+const GRAPHQL_VERSION = process.env.GRAPHQL_VERSION || graphqlMajorDefault.toString();
+const GRAPHQL_REQUIRE = GRAPHQL_VERSION === graphqlMajorDefault.toString() ? 'graphql' : `graphql-v${GRAPHQL_VERSION}`;
+process.env.GRAPHQL_REQUIRE = GRAPHQL_REQUIRE;
+
+if (GRAPHQL_REQUIRE !== 'graphql') {
+  mock('graphql', GRAPHQL_REQUIRE);
+
+  mock('graphql/language', `${GRAPHQL_REQUIRE}/language`);
+  mock('graphql/execution', `${GRAPHQL_REQUIRE}/execution`);
+  mock('graphql/utilities', `${GRAPHQL_REQUIRE}/utilities`);
+  mock('graphql/jsutils', `${GRAPHQL_REQUIRE}/jsutils`);
+  mock('graphql/validation', `${GRAPHQL_REQUIRE}/validation`);
+  mock('graphql/type', `${GRAPHQL_REQUIRE}/type`);
+  mock('graphql/polyfills', `${GRAPHQL_REQUIRE}/polyfills`);
+  mock('graphql/subscription', `${GRAPHQL_REQUIRE}/subscription`);
+  mock('graphql/error', `${GRAPHQL_REQUIRE}/error`);
```

Comment:
This might break in interesting, non-obvious ways at some point when a graphql release comes with a different set of sub packages. But I guess that is an acceptable risk.

Reply:
Agree. Lets see if this will create trouble in the future :)